instagram-app/src/components/PostContainer/PostsPage.jsx

```diff
@@ -0,0 +1,107 @@
+import '../../App.css';
+import React, { Component } from 'react'
+import Navigation from '..//SearchBar/SearchBar'
+import data from '../../dummy-data'
+import Posts from '../PostContainer/PostContainer'
+import {IconContext} from 'react-icons'
+import $ from 'jquery'
+import 'bootstrap/js/dist/modal'
+import CommentModal from '../Modals/CommentModal'
+import Fuzzy from 'fuzzy'
+
+
+export default class PostsPage extends Component {
+    constructor(props){
+        super(props)
+        this.state={
+          searching:'',
+          data:[],
+          user:'AshenPheonix',
+          newComment:'',
+          current:[]
+        }
+      }
+    
+      searchChange=e=>{
+        let force=e.target.value
+        let filter=Fuzzy.filter(force,this.state.data,{extract:i=>i.username});
+        console.log(filter);
+        this.setState(prevState=>({
+          searching:force,
+          current:prevState.data.filter(i=>i.username.includes(force))
+        }))
+      }
+    
+      componentDidMount(){
+        if(localStorage.getItem('data')){
+          let temp=JSON.parse(localStorage.getItem('data'))
+          this.setState({data:temp,current:temp,user:localStorage.getItem('username')})
+        }
+        else
+          this.setState({data:data,current:data})
+      }
+    
+      componentDidUpdate(){
+        if(this.state.current!==this.state.data && this.state.searching==='')
+          this.setState({current:this.state.data})
+          
+        localStorage.setItem('data',JSON.stringify(this.state.data))
+      }
+    
+      render(){
+        return (
+          <div className="App container">
+            <CommentModal
+              add={this.add}
+              text={this.state.newComment}
+              change={this.commentChange}
+            />
+            <IconContext.Provider value={{size:'1.5em'}}>
+              <Navigation 
+                searchValue={this.state.searching}  
+                searchEdit={this.searchChange}
+                searchFor={this.search}
+                logout={this.props.logout}
+              />
+              <Posts 
+                data={this.state.current} 
+                like={this.like} 
+                comment={this.comment}/>
+            </IconContext.Provider>
+          </div>
+        );
+      }
+    
+      like=data=>{
+        this.setState((prevState, props) => ({  data:prevState.data.map( d => d.id===data?{...d,likes:d.likes+1}:d),
+                                                current:prevState.data.map( d => d.id===data?{...d,likes:d.likes+1}:d)}))
+      }

```

Comment:
Ahh, very interesting solution to make it so that the likes are toggled rather than infinite. 

Would a logicall short-circuit not work here to skip the `:d` which simply returns the original item?